packages/catalog-realm/catalog-app/catalog.gts

```diff
@@ -376,55 +375,141 @@ class Isolated extends Component<typeof Catalog> {
+      []) as Category[];
+
+    if (!categoryInstances) {
       return [];
     }
 
-    const iconMap: Record<string, typeof IconComponent> = {
-      All: LayoutGridPlusIcon,
-      'Accounting & Finance': BuildingBank,
-      Business: BuildingIcon,
-      'Content Management': UsersIcon,
-      'Games and Entertainment': WorldIcon,
-      'Health and Fitness': HealthRecognition,
-      'Travel and Lifestyle': ShipWheelIcon,
+    // Define sphere names as a union type for better type safety
+    type SphereName = 'BUILD' | 'LEARN' | 'LIFE' | 'PLAY' | 'WORK';
+    // Define which icon to use for each sphere
+    const sphereIconMap: Record<SphereName, typeof IconComponent> = {
+      BUILD: BuildingIcon,
+      LEARN: UsersIcon,
+      LIFE: HealthRecognition,
+      PLAY: WorldIcon,
+      WORK: BuildingBank,
     };
 
-    // Create nested structure with "All" as parent
-    const nestedCategories = instances.map((instance) => {
-      return {
-        id: instance.id,
-        displayName: instance.name,
-        icon: iconMap[instance.name] || LayoutGridPlusIcon,
+    // Define the structure for sphere groups
+    interface SphereGroupSignature {
+      id: string;
+      displayName: string;
+      icon: typeof IconComponent;
+      filters: CategoryFilterItemSignature[];
+    }
+
+    // Define the structure for category filter items
+    interface CategoryFilterItemSignature {
+      id: string;
+      displayName: string;
+      icon: typeof IconComponent;
+      sphere?: string;
+    }
+
+    // Group categories by their sphere
+    const categoriesGroupedBySphere: Record<string, SphereGroupSignature> = {};
+
+    // Loop through each category and organize them by sphere
+    for (const category of categoryInstances) {
+      if (!category.sphere?.name) {
+        continue;
+      }
+
+      const sphereName = category.sphere.name;
+
+      if (!categoriesGroupedBySphere[sphereName]) {
+        categoriesGroupedBySphere[sphereName] = {
+          id: sphereName.toLowerCase(),
+          displayName: sphereName,
+          icon: sphereIconMap[sphereName as SphereName] || CategoryIcon,
+          filters: [],
+        };
+      }
+
+      const categoryFilterItem: CategoryFilterItemSignature = {
+        id: category.id,
+        displayName: category.name,
+        icon: CategoryIcon,
+        sphere: sphereName,
       };
-    });
 
-    return [
+      categoriesGroupedBySphere[sphereName].filters.push(categoryFilterItem);
+    }
+
+    // Sort categories within each sphere group
+    for (const sphereGroup of Object.values(categoriesGroupedBySphere)) {
+      sphereGroup.filters.sort((a, b) =>
+        a.displayName.localeCompare(b.displayName),
+      );
+    }
+
+    // Create the final list with "All" button at the top
+    // Sort categories by their display name
+    const allCategoriesList: (
+      | SphereGroupSignature
+      | CategoryFilterItemSignature
+    )[] = [
```

Comment:
Im confused with this type? 

SphereGroupSignature

and 

CategoryFilterItemSignature

look absolutely different